hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -241,34 +244,45 @@ private void approveAllowance(
 
         /* --- Apply changes to state --- */
         final var allowanceMaxAccountLimit = hederaConfig.allowancesMaxAccountLimit();
-        applyCryptoAllowances(cryptoAllowances, payerId, accountStore, allowanceMaxAccountLimit);
-        applyFungibleTokenAllowances(tokenAllowances, payerId, accountStore, allowanceMaxAccountLimit);
+        applyCryptoAllowances(
+                cryptoAllowances, payerId, accountStore, allowanceMaxAccountLimit, context.expiryValidator());
+        applyFungibleTokenAllowances(
+                tokenAllowances, payerId, accountStore, allowanceMaxAccountLimit, context.expiryValidator());
         applyNftAllowances(
-                nftAllowances, payerId, accountStore, tokenStore, uniqueTokenStore, allowanceMaxAccountLimit);
+                nftAllowances,
+                payerId,
+                accountStore,
+                tokenStore,
+                uniqueTokenStore,
+                allowanceMaxAccountLimit,
+                context.expiryValidator());
```

Comment:
nit : we could just declare expiryValidator outside as `context.expiryValidator()` and use it in all three places

Reply:
Need not be done in this PR.

Reply:
Done!